app/lib/package/backend.dart

```diff
@@ -1466,3 +1474,52 @@ void checkPackageVersionParams(String package, [String? version]) {
     }
   }
 }
+
+/// Helper utility class for interfacing with Cloud Storage for storing
+/// images.
+class ImageStorage {
+  final Bucket bucket;
+  ImageStorage(this.bucket);
+
+  final _supportedMIMETypes = [
+    'image/apng',
+    'image/avif',
+    'image/gif',
+    'image/jpeg',
+    'image/png',
+    'image/svg+xml',
+    'image/webp'
+  ];
+
+  /// Uploads an image to image storage, at '<package>/<version>/<imageFileName>'.
+  ///
+  /// Throws an [ImageRejectedException] if the provided image file is not one of
+  /// the following supported MIME types : image/apng, image/avif, image/gif,
+  /// image/jpeg, image/png, image/svg+xml, image/webp.
+  Future<void> upload(
+      String package,
+      String version,
+      Stream<List<int>> Function() openStream,
+      String imageFilePath,
+      int length) async {
+    final String? mimeType = lookupMimeType(imageFilePath);
+
+    if (mimeType == null || !_validateImage(mimeType)) {
+      _logger.info(
+          'Image upload of $imageFilePath for package $package failed. '
+          'The file $imageFilePath is not one of the following supported MIME types:'
+          ' ${_supportedMIMETypes.join(', ')}.');
+      throw ImageRejectedException.invalid();
+    }
+
+    return bucket.uploadPublic(
+        [package, version, imageFilePath.split('/').last].join('/'),
```

Comment:
`imageFilePath.split('/')` why? we can just have the entire path to the image file within the package...

Doesn't this just invite conflicts?

Reply:
Good point, let's keep the entire path.